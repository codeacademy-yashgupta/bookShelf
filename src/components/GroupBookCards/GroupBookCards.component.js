import React, { Component } from 'react';
import './GroupBookCards.css';
import BookCard from '../BookCard/BookCard.component';

class GroupBookCards extends Component {
  static defaultProps = {
    authorRows: [{
      Author: 'J K Rowling', id: 10, Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', rating: 4.45,
    }, {
      Author: 'J K Rowling', id: 20, Name: 'Harry Potter and the Chamber of Secrets (Harry Potter, #2)', rating: 4.38,
    }, {
      Author: 'J K Rowling', id: 30, Name: 'Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)', rating: 4.54,
    }, {
      Author: 'J K Rowling', id: 40, Name: 'Harry Potter and the Goblet of Fire (Harry Potter, #4)', rating: 4.53,
    }, {
      Author: 'J K Rowling', id: 50, Name: 'Harry Potter and the Order of the Phoenix (Harry Potter, #5)', rating: 4.47,
    }, {
      Author: 'J K Rowling', id: 60, Name: 'Harry Potter and the Half-Blood Prince (Harry Potter, #6)', rating: 4.54,
    }, {
      Author: 'J K Rowling', id: 70, Name: 'Harry Potter and the Deathly Hallows (Harry Potter, #7)', rating: 4.62,
    }],

  }

  render() {
    const { authorRows } = this.props;
    const rows = [];
    authorRows.forEach((bookCard, index) => {
      rows.push(<BookCard />);
    });
    return (
      <div className="group-by-author">
        <fieldset>
          <legend>J.K Rowling</legend>
          <div className="all-cards">
            {rows}
          </div>
        </fieldset>
      </div>
    );
  }
}

export default GroupBookCards;
