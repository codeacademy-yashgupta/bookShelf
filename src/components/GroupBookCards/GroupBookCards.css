.group-by-author legend{
  background-color: white;
  font-size: 30px;
  padding:5px;
  font-style: italic;
  font-weight: bold;
  color: orange;
  text-align: center
}

fieldset {
  display: flex;
  border: 8px solid orange;
  border-radius: 50px;
  width: 100%;
}

.group-by-author {
  display: flex;
  width: 80%;
  justify-content: center;
  margin: auto;
  margin-top: 30px;
  flex-direction: row
}
.all-cards {
  display: flex;
  justify-content: space-evenly;
  flex-direction: row;
  width: 100%;
  flex-wrap: wrap;

}