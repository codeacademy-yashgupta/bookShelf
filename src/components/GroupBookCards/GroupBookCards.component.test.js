import React from 'react';
import renderer from 'react-test-renderer';
import GroupBookCards from './GroupBookCards.component';

describe('GroupBookCards', () => {
  it('should render Group books card', () => {
    const tree = renderer.create(<GroupBookCards />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
