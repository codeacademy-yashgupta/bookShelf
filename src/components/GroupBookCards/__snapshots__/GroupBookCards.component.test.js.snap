// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupBookCards should render Group books card 1`] = `
<div
  className="group-by-author"
>
  <fieldset>
    <legend />
    <div
      className="all-cards"
    >
      <div
        className="card-body"
      >
        <div
          className="card-image"
        >
          <img
            alt=""
            src="./images/cover_image.jpeg"
          />
        </div>
        <div
          className="card-text-title"
        >
          Harry Potter and the Sorcerers Stone (Harry Potter, #1)
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-text-rating"
          >
            4.45
          </div>
          <div
            className="card-like"
          >
            <div
              className="card-reaction"
            >
              <img
                alt="Press for like"
                className="icon-heart"
                onClick={[Function]}
                src="./icons/heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="card-body1"
      >
        <div
          className="card-image"
        >
          <img
            alt=""
            src="./images/cover_image.jpeg"
          />
        </div>
        <div
          className="card-text-title"
        >
          Harry Potter and the Chamber of Secrets (Harry Potter, #2)
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-text-rating"
          >
            4.38
          </div>
          <div
            className="card-like"
          >
            <div
              className="card-reaction"
            >
              <img
                alt="Press for like"
                className="icon-heart"
                onClick={[Function]}
                src="./icons/heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <div
          className="card-image"
        >
          <img
            alt=""
            src="./images/cover_image.jpeg"
          />
        </div>
        <div
          className="card-text-title"
        >
          Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-text-rating"
          >
            4.54
          </div>
          <div
            className="card-like"
          >
            <div
              className="card-reaction"
            >
              <img
                alt="Press for like"
                className="icon-heart"
                onClick={[Function]}
                src="./icons/heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="card-body1"
      >
        <div
          className="card-image"
        >
          <img
            alt=""
            src="./images/cover_image.jpeg"
          />
        </div>
        <div
          className="card-text-title"
        >
          Harry Potter and the Goblet of Fire (Harry Potter, #4)
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-text-rating"
          >
            4.53
          </div>
          <div
            className="card-like"
          >
            <div
              className="card-reaction"
            >
              <img
                alt="Press for like"
                className="icon-heart"
                onClick={[Function]}
                src="./icons/heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <div
          className="card-image"
        >
          <img
            alt=""
            src="./images/cover_image.jpeg"
          />
        </div>
        <div
          className="card-text-title"
        >
          Harry Potter and the Order of the Phoenix (Harry Potter, #5)
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-text-rating"
          >
            4.47
          </div>
          <div
            className="card-like"
          >
            <div
              className="card-reaction"
            >
              <img
                alt="Press for like"
                className="icon-heart"
                onClick={[Function]}
                src="./icons/heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="card-body1"
      >
        <div
          className="card-image"
        >
          <img
            alt=""
            src="./images/cover_image.jpeg"
          />
        </div>
        <div
          className="card-text-title"
        >
          Harry Potter and the Half-Blood Prince (Harry Potter, #6)
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-text-rating"
          >
            4.54
          </div>
          <div
            className="card-like"
          >
            <div
              className="card-reaction"
            >
              <img
                alt="Press for like"
                className="icon-heart"
                onClick={[Function]}
                src="./icons/heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <div
          className="card-image"
        >
          <img
            alt=""
            src="./images/cover_image.jpeg"
          />
        </div>
        <div
          className="card-text-title"
        >
          Harry Potter and the Deathly Hallows (Harry Potter, #7)
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-text-rating"
          >
            4.62
          </div>
          <div
            className="card-like"
          >
            <div
              className="card-reaction"
            >
              <img
                alt="Press for like"
                className="icon-heart"
                onClick={[Function]}
                src="./icons/heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;
